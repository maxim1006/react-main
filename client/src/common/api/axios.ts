import axios, { AxiosStatic } from 'axios';

const customAxios = axios.create({
    baseURL: 'http://localhost:3001/api',
    // headers: {}
}) as AxiosStatic;

customAxios['CancelToken'] = axios.CancelToken;
customAxios['isCancel'] = axios.isCancel;

export default customAxios;

// пример кастомного error handling
// const ErrorDescription = ({ error }) => {
//     return (
//         <>
//             Please try again later <br /> {error}
//     </>
// );
// };
//
// customAxios.interceptors.response.use(
//     function (response) {
//         // Do something with response data
//         return response;
//     },
//     function (error) {
//         notification['error']({
//             message: 'Something went wrong',
//             description: <ErrorDescription error={error.message} />,
//     });
//         // Do something with response error
//         return Promise.reject(error);
//     }
// );
// также с request
// axios.interceptors.request.use(function (config) {
//     // Do something before request is sent
//     return config;
// }, function (error) {
//     // Do something with request error
//     return Promise.reject(error);
// });
