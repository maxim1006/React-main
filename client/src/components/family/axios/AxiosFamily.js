import React, { memo } from "react";
import MaterialLoader from "../../loader/MaterialLoader";
import useGetRequest from "../../hooks/useGetRequest";

export default memo(() => {
    const family = useGetRequest({ url: "family" });

    return family ? (
        <ul>
            {family.map(({ name, id, age }) => (
                <li key={id}>
                    {name}
:{age}
                </li>
            ))}
        </ul>
    ) : (
        <MaterialLoader />
    );
});
