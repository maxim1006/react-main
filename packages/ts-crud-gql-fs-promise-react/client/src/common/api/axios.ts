import axios from 'axios';

const customAxios: any = axios.create({
    // baseURL: "http://localhost:3001/api"
    // headers: {}
});

customAxios.CancelToken = axios.CancelToken;
customAxios.isCancel = axios.isCancel;

export default customAxios;
