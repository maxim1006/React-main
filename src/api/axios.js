import axios from "axios";

const ciustomAxios = axios.create({
   baseURL: "http://localhost:3001/api"
});

ciustomAxios.CancelToken = axios.CancelToken;
ciustomAxios.isCancel = axios.isCancel;

export default ciustomAxios;
