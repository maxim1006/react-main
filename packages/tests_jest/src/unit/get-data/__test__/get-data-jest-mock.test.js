// как только мокаю эти пакеты сразу ругань будет // Cannot read properties of undefined (reading 'then')
import axios from 'axios';
import getData, { getDataAxios } from '../get-data';

jest.mock('node-fetch');
jest.mock('axios');

describe('Get Data', () => {
    let res;

    beforeAll(() => {
        res = {
            data: { userId: 1, id: 1, title: 'delectus aut autem', completed: false },
        };
    });

    test('Get data fetch empty url', async () => {
        // так ну тут замес в том что мокаю запросы axios и считаю количество вызовов + соответсвтие на возвращаемый результат
        axios.get.mockResolvedValue(res);
        const data = await getDataAxios();
        expect(axios.get).toBeCalledTimes(1);
        expect(data).toEqual(res);
        // эта тема сохраняет snapshot запроса чтобы если кто-то поменяет res тест упал
        expect(data).toMatchSnapshot(res);
    });
});
