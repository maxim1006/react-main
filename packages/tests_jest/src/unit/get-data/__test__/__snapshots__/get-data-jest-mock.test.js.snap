// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get Data Get data fetch empty url 1`] = `
Object {
  "data": Object {
    "completed": false,
    "id": 1,
    "title": "delectus aut autem",
    "userId": 1,
  },
}
`;
